libgadget/blackhole.c

```diff
@@ -141,7 +180,8 @@ static double blackhole_soundspeed(double entropy, double pressure, double rho)
     return cs;
 }
 
-void blackhole(ForceTree * tree)
+void
+blackhole(ForceTree * tree, double * TimeNextSeedingCheck)
```

Comment:
Wht about spliting to two arguments: (double TimeSeedingCheck, double * TimeNextSeedingCheck)? Then the first one is a const and the second one is out-only.

Reply:
I tried that first, with blackhole returning the time of the next seeding check but it looked sort of ugly. I'll add a comment.